atom/lib/import-js.js

```diff
@@ -0,0 +1,302 @@
+        * @param {String} data
+        */
+        stdout(data) {
+          result.output = data;
+        },
+
+        /**
+        * @param {Number} code
+        */
+        exit(code) {
+          if (code === 0) {
+            resolve(result);
+          }
+          reject(new Error(`Exited with code ${code}`), result);
+        },
+      };
+
+      const spawnedProcess = new BufferedProcess(processOptions);
+      spawnedProcess.onWillThrowError((error) => {
+        atom.notifications.addError('import-js', {
+          detail: error,
+        });
+        reject(error, result);
+      });
+
+      const currentFileContents = editor.getText();
+      spawnedProcess.process.stdin.write(currentFileContents);
+      spawnedProcess.process.stdin.end();
+    });
+  }); // new Promise
+}
+
+/**
+ * @return {String} The word under the cursor
+ */
+function getCurrentWord() {
+  const editor = atom.workspace.getActiveTextEditor();
+  if (!editor) {
+    return undefined;
+  }
+
+  const cursor = editor.getLastCursor();
+  if (!cursor) {
+    return undefined;
+  }
+
+  const wordRange = cursor.getCurrentWordBufferRange();
+  return editor.getTextInBufferRange(wordRange);
+}
+
+/**
+ * @param {Array<Object>} selectionsToAskFor
+ * @param {Array<Object>} previousSelections
+ */
+function askForSelections(selectionsToAskFor, previousSelections) {
+  const selections = previousSelections || [];
+
+  if (selectionsToAskFor.length === 0) {
+    // We have no more selections to ask for, so we can wrap it up!
+    if (selections.length === 0) {
+      // No selections were chosen.
+      return undefined;
+    }
+
+    // Foo:0,Bar:1
+    const selectionsString = selections
+      .map((selection) => `${selection.word}:${selection.alternativeIndex}`)
+      .join(',');
+
+    /* eslint-disable no-use-before-define */
+    importWord(selectionsString);
+    /* eslint-enable no-use-before-define */
+    return undefined;
+  }
+
+  // Ask for the next selection
+  const selectionToAskFor = selectionsToAskFor.shift();
+
+  // TODO show what word we are asking for
+  const askView = new AskForSelectionView();
+  askView.setItems(selectionToAskFor.alternatives);
+  askView.show();
+  askView.deferred
+    .then((alternativeIndex) => {
+      const selection = {
+        word: selectionToAskFor.word,
+        alternativeIndex,
+      };
+
+      return selections.concat([selection]);
+    })
+    // Selection was cancelled, so let's just move on.
+    .catch(() => selections)
+    .then((newSelections) => {
+      askView.destroy();
+      askForSelections(selectionsToAskFor, newSelections);
+    });
+
+  return undefined;
+}
+
+/**
+ * @param {String} selections
+ */
+function importWord(selections) {
+  const args = [];
+
+  if (selections) {
+    // We have selections to import.
+    args.push('--selections', selections);
+  } else {
+    // There are no selections to import, so let's figure out what the current
+    // word is.
```

Comment:
I just tried it out and I don't see a bug. Since the selections contains the words that need to be imported and their indexes in the list, I don't see why `--word` is needed. And actually, instead of calling `importWord` with the selections, we should really just call a new function `importSelections`.